velox/common/memory/SharedArbitrator.h

```diff
@@ -35,7 +35,52 @@ namespace facebook::velox::memory {
 /// partial aggregation or persistent shuffle data flushes.
 class SharedArbitrator : public memory::MemoryArbitrator {
  public:
-  explicit SharedArbitrator(const Config& config);
+  struct SharedArbitratorConfig {
+    /// The memory capacity reserved to ensure each running query has minimal
+    /// capacity of 'memoryPoolReservedCapacity' to run.
+    static constexpr std::string_view kReservedCapacity{"reserved-capacity"};
```

Comment:
We might not need this structure but a simple list of public extra config strings? Or called it SharedArbitrator::ExtraConfig?

Reply:
We need both literal string and defaults? So maybe I can change the name to ExtraConfig